scrapers/al/events.py

```diff
@@ -13,59 +14,62 @@ class ALEventScraper(Scraper, LXMLMixin):
+
+        headers = {
+            "Accept": "*/*",
+            "Accept-Language": "en-US,en;q=0.9",
+            "Authorization": "Bearer undefined",
+            "Content-Type": "application/json",
+            "Origin": "https://alison.legislature.state.al.us",
+            "Referer": "https://alison.legislature.state.al.us/",
+        }
+
+        # start from the first of the current month
+        from_date = datetime.datetime.today().replace(day=1).strftime("%Y-%m-%d")
+        query = (
+            '{hearingsMeetings(eventType:"meeting", body:"", keyword:"", toDate:"3000-02-06", '
+            f'fromDate:"{from_date}", sortTime:"", direction:"ASC", orderBy:"SortTime", )'
+            "{ EventDt,EventTm,Location,EventTitle,EventDesc,Body,DeadlineDt,PublicHearing,"
+            "Committee,AgendaUrl,SortTime,OidMeeting }}"
         )
-        event_count = 0
-        for row in rows[1:]:
-            date = row.xpath("td")[0].text_content().strip()
-            time = row.xpath("td")[1].text_content().strip()
-            details = row.xpath("td")[4].text_content().strip()
-
-            if time != "":
-                date_with_time = "{} {}".format(date, time)
-            else:
-                # sometimes the time is the first part of the decription
-                match = re.match(r"\s*\d+:\d+\s*[ap]m", details, flags=re.I)
-                if match:
-                    date_with_time = "{} {}".format(date, match.group())
-
-            location = row.xpath("td")[2].text_content().strip()
 
-            if re.match(r"^Room", location, flags=re.IGNORECASE):
-                location = f"{location}, 11 South Union Street, Montgomery, Alabama, United States"
+        json_data = {
+            "query": query,
+            "operationName": "",
+            "variables": [],
+        }
 
-            # 11 South Union Street, Montgomery, Alabama, United States
-            # TODO: IF location is "room (X)" add state house
-            # TODO: REplace "state house" with address
+        page = self.post(gql_url, headers=headers, json=json_data)
+        page = json.loads(page.content)
 
-            # 32°22′37.294″N 86°17′57.991″W
+        if len(page["data"]["hearingsMeetings"]) == 0:
+            raise EmptyScrape
 
-            # host = row.xpath('td')[3].text_content().strip()
-            name = row.xpath("td")[3].text_content().strip()
-            if name == "":
-                continue
+        for row in page["data"]["hearingsMeetings"]:
+            event_date = self._TZ.localize(dateutil.parser.parse(row["SortTime"]))
+            event_title = row["EventTitle"]
+            event_location = row["Location"]
+            if event_location.startswith("Room"):
+                event_location = (
+                    f"11 South Union St, Montgomery, AL 36130. {event_location}"
+                )
+            event_desc = row["EventDesc"]
 
             event = Event(
```

Comment:
Probably go ahead and add duplicate detection?

Reply:
Do I need to do anything other than creating/setting the event.dedupe_key for that?